Organization/Organization/MainViewController.m

```diff
@@ -1,57 +1,70 @@
 #import "MainViewController.h"
 #import "DetailViewController.h"
 #import "CreateEmployeeViewController.h"
-#import "Employee.h"
-#import "Organization.h"
+#import "Employee+CoreDataClass.h"
+#import "Organization+CoreDataClass.h"
+#import "AppDelegate.h"
 
-@interface MainViewController ()		
+@interface MainViewController ()
 
 @end
 
 @implementation MainViewController
 
+@synthesize org;
+@synthesize managedObjectContext = _managedObjectContext;
+@synthesize managedObjectModel = _managedObjectModel;
+@synthesize persistentStoreCoordinator = _persistentStoreCoordinator;
+
+NSString* attributeValue = @"Siemens";
+NSString* attributeName = @"name"; //genious constant
+
+
 - (void)viewDidLoad
 {
     [super viewDidLoad];
-    
-    Organization *org = [[Organization alloc] initWithName:@"Inter"];
-    for(int i = 0; i < 4; i++)
-    {
-        NSString *name = [@(i).stringValue stringByAppendingString: @"vasya "];
-        NSString *fullName = [name stringByAppendingString:@"pupkin"];
-        [org addEmployeeWithName:fullName];
-    }
-    
-    self.org = org;
-    
+  
+    NSManagedObjectContext *context = [AppDelegate context];
+    NSFetchRequest *request = [[NSFetchRequest alloc] init];
+    NSEntityDescription *entity = [NSEntityDescription entityForName:@"Organization" inManagedObjectContext:context];
+    [request setEntity:entity];
+    org = [context executeFetchRequest:request error:nil] == nil ?
+                [context executeFetchRequest:request error:nil] [0] :
+                [self primaryInit:context];
+}
+
+- (Organization *)primaryInit:(NSManagedObjectContext *)context
+{
+    NSFetchRequest *request = [[NSFetchRequest alloc] init];
+    Organization *organization = [NSEntityDescription insertNewObjectForEntityForName:@"Organization" inManagedObjectContext:context];
+    organization.name = attributeValue;
+    [AppDelegate saveContext];
+    NSEntityDescription *entity = [NSEntityDescription entityForName:@"Organization" inManagedObjectContext:context];
+    [request setEntity:entity];
+    return [context executeFetchRequest:request error:nil][0];
```

Comment:
Why do you need to fetch organization from DB, if you already have it locally in organization variable?